runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java

```diff
@@ -971,7 +987,9 @@ public void translateNode(
               .transform(fullName, outputTypeInfo, (OneInputStreamOperator) doFnOperator)
               .uid(fullName);
 
-      context.setOutputDataStream(context.getOutput(transform), outDataStream);
+      final PCollection<KV<K, Iterable<InputT>>> output = context.getOutput(transform);
+      context.setOutputDataStream(output, outDataStream);
+      context.setProducer(output, transform);
```

Comment:
It would be nice if these explicit calls wouldn't be required. I believe `context.setOutputDataStream` internally has the current transform available. So we could perform it internally in the context.

Reply:
nice catch 👍